### PR TITLE
Fix language-only VLM LoRA export names for inference

### PR DESCRIPTION
**Pull request by mattpeck-aq**
Fixes a silent LoRA weight-loading failure when FSDP trains a VLM in language-only mode: exported adapter names omit the language_model. namespace expected by the inference model.

Shares the existing full-weight sync prefix rule with LoRA export, inserting it inside PEFT’s wrapper. Unrecognized prefixes remain unchanged.

image
image

---

> [!NOTE]
> **Medium Risk**
> Changes inference handoff naming for language-only VLM LoRA (training/inference sync path); wrong logic would still fail silently at adapter load, but scope is narrow and covered by new tests.
> 
> **Overview**
> Fixes **silent LoRA load failures** when FSDP trains a vision-language model with `trainer.policy.language_model_only=true`: exported adapter keys now include the `language_model.` namespace vLLM expects, matching full-weight sync behavior.
> 
> Introduces shared helper **`get_inference_weight_prefix`** and applies it in **`_save_lora_adapters_and_sync`**, inserting the prefix inside PEFT’s `base_model.model.` wrapper for standard keys while **leaving unrecognized parameter names unchanged**. Docs add a short note on text-only VLM LoRA training; new vLLM/Qwen3.5 tests cover export naming for full sync and edge-case prefixes.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit d743ad0a4aadf394b5bc3217093e670d315a7b83. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: skyrl/backends/skyrl_train/workers/fsdp/fsdp_worker.py [hunk 1]
```diff
@@ -48,6 +48,16 @@
     from skyrl.train.config.config import InferenceEngineConfig
 
 
+def _get_inference_weight_prefix(is_multimodal_lm_only: bool) -> str:
```
Comment (erictang000) [inline skyrl/backends/skyrl_train/workers/fsdp/fsdp_worker.py]: can we move this helper to workers/worker_utils.py to keep things clean?

**Review by erictang000 — APPROVED** [review]
